server/lib/matchmaking/models.ts

```diff
@@ -433,6 +447,45 @@ export async function getMatchmakingRatingsForUser(
   }
 }
 
+/**
+ * Returns a user's finalized rank for all seasons and matchmaking type combos. Rank for the current
+ * season, as well as any season and matchmaking type combo that the user has not completed a game
+ * in will not be included in the result. The results are ordered by the start date of the season.
+ */
+export async function getMatchmakingFinalizedRanksForUser(
+  userId: SbUserId,
+  withClient?: DbClient,
+): Promise<FinalizedMatchmakingRank[]> {
+  const { client, done } = await db(withClient)
+
+  try {
+    const result = await client.query<DbMatchmakingFinalizedRank>(sql`
+      SELECT mfr.user_id, mfr.season_id, mfr.matchmaking_type, mfr.rank, mr.rating, mr.points,
+        mr.bonus_used, mr.lifetime_games, mr.wins, mr.losses,
+        mr.p_wins, mr.p_losses,
+        mr.t_wins, mr.t_losses,
+        mr.z_wins, mr.z_losses,
+        mr.r_wins, mr.r_losses,
+        mr.r_p_wins, mr.r_p_losses, mr.r_t_wins, mr.r_t_losses, mr.r_z_wins, mr.r_z_losses,
+        mr.last_played_date
+      FROM
+        matchmaking_finalized_ranks mfr
+        INNER JOIN matchmaking_ratings mr ON mfr.season_id = mr.season_id
+          AND mfr.user_id = mr.user_id
+          AND mfr.matchmaking_type = mr.matchmaking_type
+        INNER JOIN matchmaking_seasons ms ON mfr.season_id = ms.id
+      WHERE
+        mfr.user_id = ${userId}
+      ORDER BY
+        ms.start_date DESC
```

Comment:
Can we have an explicit ordering for the matchmaking types within the same season as well?